File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-1.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> map = new LinkedHashMap<>();

        for (String s : arr){
            if (!map.containsKey(s)){
                map.put(s, 1);
            } else {
                map.put(s, map.get(s) + 1);
            }
        }

        for (String key : map.keySet()) {
            if (map.get(key) == 1) {
                k--;
                if (k == 0) {
                    return key;
                }
            }
        }

        return "";
    }
}

File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-1.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        k = [0 for _ in range(501)]
        for x in arr:
            k[x] += 1

        for i in range(500, 0, -1):
            if i == k[i]:
                return i
        return -1